Python Updates 2022-06-01 (was: python3: 3.9 -> 3.10)

mweinelt opened the pull request:
First staging run after the 22.05 release. During this cycle we update `python3` from 3.9 to 3.10, while still recursing into both versions.

Reopened on python-updates so we have a jobset where we can build the resulting changes.

https://hydra.nixos.org/jobset/nixpkgs/python-updates

fabaff committed:
python310Packages.pure-eval: 0.2.1 -> 0.2.2

SuperSandro2000 committed:
python310Packages.hypothesis: 6.40.0 -> 6.46.3

fabaff committed:
python310Packages.aiomysql: 0.1.0 -> 0.1.1

fabaff committed:
python310Packages.pytest-httpbin: 1.0.1 -> 1.0.2

fabaff committed:
python310Packages.pytest-localserver: 0.5.1.post0 -> 0.6.0

fabaff committed:
python310Packages.sopel: 7.1.8 -> 7.1.9

fabaff committed:
python310Packages.pytest-randomly: 3.11.0 -> 3.12.0

fabaff committed:
python310Packages.vulture: 2.3 -> 2.4

https://github.com/jendrikseipp/vulture/releases/tag/v2.4

fabaff committed:
python310Packages.gcsfs: 2022.3.0 -> 2022.5.0

fabaff committed:
python310Packages.dask-gateway-server: 0.9.0 -> 2022.4.0

fabaff committed:
python310Packages.distributed: 2022.2.1 -> 2022.5.0

fabaff committed:
python310Packages.dask: 2022.02.1 -> 2022.05.0

fabaff committed:
python310Packages.locket: 0.2.1 -> 1.0.0

fabaff committed:
python310Packages.s3fs: 2022.2.0 -> 2022.5.0

fabaff committed:
python310Packages.tifffile: 2022.3.25 -> 2022.5.4

fabaff committed:
python310Packages.imageio: 2.16.1 -> 2.19.2

fabaff committed:
python310Packages.GitPython: 3.1.25 -> 3.1.27

fabaff committed:
python310Packages.apipkg: 2.1.0 -> 2.1.1

mweinelt committed:
python3: 3.9 -> 3.10

Defaults python3 to Python 3.10 while still keeping Python 3.9 around
until after the 22.11 release.

mweinelt committed:
doc/python: update python version references

fabaff committed:
python310Packages.pytest-subtests: 0.7.0 -> 0.8.0

fabaff committed:
python310Packages.pyarrow: disable flaky test

fabaff committed:
python310Packages.dask: 2022.05.0 -> 2022.05.2

fabaff committed:
python310Packages.pyzmq: 22.3.0 -> 23.0.0

fabaff committed:
python310Packages.distributed: 2022.5.0 -> 2022.5.2

SuperSandro2000 committed:
python310Packages.botocore: drop ordereddict

SuperSandro2000 committed:
python310Packages.flask-limiter: drop ordereddict

SuperSandro2000 committed:
python310Packages.ordereddict: drop

SuperSandro2000 committed:
python310Packages.yamlordereddictloader: drop ordereddict

SuperSandro2000 committed:
python27Packages.botocore: drop ordereddict

SuperSandro2000 committed:
jira_cli: drop ordereddict

SuperSandro2000 committed:
python310Packages.pyscss: drop ordereddict

SuperSandro2000 committed:
python310Packages.pyscss: switch to pytestCheckHook

fabaff committed:
python310Packages.webob: switch to pytestCheckHook

mweinelt committed:
python3Minimal: 3.9 -> 3.10

SuperSandro2000 committed:
python310Packages.raven: update dependencies, description, comment

SuperSandro2000 committed:
python310Packages.httpbin: fix dependencies

SuperSandro2000 committed:
python310Packages.urllib3: add optional-dependencies

fabaff committed:
python310Packages.streamz: disable flaky tests

fabaff committed:
python310Packages.pyarrow: disable flaky test

SuperSandro2000 committed:
python310Packages.lxml: 4.8.0 -> 4.9.0

fabaff committed:
python310Packages.intake: 0.6.4 -> 0.6.5

SuperSandro2000 committed:
python310Packages.astroid: 2.11.2 -> 2.11.4, adopt

SuperSandro2000 committed:
python310Packages.pylint: 2.13.5 -> 2.14.0, adopt

fabaff committed:
python310Packages.fsspec: 2022.3.0 -> 2022.5.0

SuperSandro2000 committed:
python310Packages.asgiref: 3.5.0 -> 3.5.2

SuperSandro2000 committed:
python310Packages.certifi: 2021.10.08 -> 2022.05.18.1

SuperSandro2000 committed:
python310Packages.cryptography: 36.0.2 -> 37.0.2

SuperSandro2000 committed:
python310Packages.hypothesis: 6.46.3 -> 6.46.10

SuperSandro2000 committed:
python310Packages.pbr: 5.8.1 -> 5.9.0

SuperSandro2000 committed:
python310Packages.libcst: 0.4.1 -> 0.4.3, drop inactive maintainer

SuperSandro2000 committed:
python310Packages.pillow: 9.1.0 -> 9.1.1

SuperSandro2000 committed:
python310Packages.typed-ast: 1.5.2 -> 1.5.4

SuperSandro2000 committed:
python310Packages.types-typed-ast: 1.5.4 -> 1.5.6

SuperSandro2000 committed:
python310Packages.xmltodict: 0.12.0 -> 0.13.0

SuperSandro2000 committed:
python310Packages.zipp: 3.7.0 -> 3.8.0

mweinelt committed:
spidermonkey_78: pin python39

Because the configure script hasn't been update to work with Python 3.10
and never will. Spidermonkey 78 is EOL.

Traceback (most recent call last):
  File "/build/firefox-78.15.0/obj/../js/src/../../configure.py", line 25, in <module>
    from mozbuild.configure import (
  File "/build/firefox-78.15.0/python/mozbuild/mozbuild/configure/__init__.py", line 33, in <module>
    from mozbuild.util import (
  File "/build/firefox-78.15.0/python/mozbuild/mozbuild/util.py", line 760, in <module>
    class HierarchicalStringList(object):
  File "/build/firefox-78.15.0/python/mozbuild/mozbuild/util.py", line 785, in HierarchicalStringList
    class StringListAdaptor(collections.Sequence):
AttributeError: module 'collections' has no attribute 'Sequence'

mweinelt committed:
python3Packages.gyp: 2020-05-12 -> unstable-2022-04-01

fabaff committed:
python310Packages.fastparquet: add optional dependency

fabaff committed:
python310Packages.s3fs: disable on older Python releases

fabaff committed:
python310Packages.cloudpickle: 2.0.0 -> 2.1.0

SuperSandro2000 committed:
python310Packages.pysqueezebox: fix formatting

SuperSandro2000 commented:
dtc is failing to build: https://hydra.nixos.org/build/178939319

https://git.kernel.org/pub/scm/utils/dtc/dtc.git/log/

One potential fix could be https://git.kernel.org/pub/scm/utils/dtc/dtc.git/commit/?id=383e148b70a47ab15f97a19bb999d54f9c3e810f

SuperSandro2000 commented on pkgs/development/python-modules/webob/default.nix:
```diff
@@ -1,25 +1,41 @@
 
-  propagatedBuildInputs = [ nose pytest ];
+  propagatedBuildInputs = [
+    pytestCheckHook
```
That is really bad. pytestCheckHook should **never** be propagated since it enforces checks even if doCheck is false.

fabaff replied:
Was not done on purpose, more an oversight on my side.

SuperSandro2000 committed:
python310Packages.cachecontrol: 0.12.10 -> 0.12.11

SuperSandro2000 committed:
python310Packages.cherrypy: properly declare optional  dependencies

SuperSandro2000 committed:
python310Packages.repoze_lru: add pythonImportsCheck

SuperSandro2000 committed:
python310Packages.routes: update meta, normalize pname, cleanup

SuperSandro2000 committed:
python310Packages.soupsieve: document circular dependency, cleanup

SuperSandro2000 committed:
dtc: fix python 3.10 compatibility

sikmir committed:
python3Packages.Rtree: 0.9.7 → 1.0.0

sikmir committed:
python3Packages.networkx: 2.7.1 → 2.8.2

sikmir committed:
python3Packages.osmnx: 1.1.2 → 1.2.0

fabaff committed:
unicorn: 2.0.0-rc5 -> 2.0.0-rc7

fabaff committed:
python310Packages.pefile: 2021.9.3 -> 2022.5.30

fabaff committed:
python310Packages.pyelftools: 0.27 -> 0.28

fabaff committed:
python310Packages.qiling: 1.4.2 -> 1.4.3

fabaff committed:
python310Packages.responses: 0.20.0 -> 0.21.0

fabaff committed:
python310Packages.cachetools: 5.0.0 -> 5.2.0

fabaff committed:
python310Packages.httpcore: 0.14.7 -> 0.15.0

mweinelt committed:
python3Packages.numpy: 1.21.5 -> 1.21.6

https://github.com/numpy/numpy/releases/tag/v1.21.6

fabaff committed:
python310Packages.httpx: 0.22.0 -> 0.23.0

fabaff committed:
python310Packages.pytest-httpx: 0.20.0 -> 0.21.0

fabaff committed:
python310Packages.python-slugify: 6.1.1 -> 6.1.2

fabaff committed:
python310Packages.simple-rest-client: 1.1.2 -> 1.1.3

SuperSandro2000 committed:
python310Packages.xdis: unstable-2022-04-13 -> 6.0.4

dotlambda committed:
python310Packages.pylink-square: 0.8.1 -> 0.13.0

https://github.com/square/pylink/blob/v0.13.0/CHANGELOG.md

SuperSandro2000 committed:
python310Packages.m2r: fix tests under python 3.10

SuperSandro2000 committed:
python310Packages.billiard: disable flaky test

mweinelt committed:
python3Packages.sqlalchemy: 1.4.36 -> 1.4.37

https://github.com/sqlalchemy/sqlalchemy/releases/tag/rel_1_4_37

SuperSandro2000 committed:
python310Packages.pgpy: ignore broken test

dotlambda committed:
sftpman: 1.1.3 -> 1.2.2

SuperSandro2000 committed:
MACS2: mark broken

SuperSandro2000 committed:
python310Packages.jsonschema: 4.5.1 -> 4.6.0

fabaff committed:
Python310Packages.sparse: disable on older Python releases

mweinelt committed:
python3Packages.moto: 3.1.3 -> 3.1.11

mweinelt committed:
python3Packages.certbot: 1.24.0 -> 1.27.0

mweinelt committed:
python3Packages.hass-nabuacasa: update dependency constraints

mweinelt committed:
home-assistant: update python-slugify override

fabaff committed:
python310Packages.dask-glm: handle optional dependencies

fabaff committed:
python310Packages.dask-ml: handle optional dependencies

fabaff committed:
python310Packages.dask: 2022.05.0 -> 2022.05.2

dotlambda committed:
python310Packages.cssutils: 2.4.0 -> 2.4.1

https://github.com/jaraco/cssutils/blob/v2.4.1/CHANGES.rst

dotlambda committed:
python310Packages.weasyprint: update dependencies

dotlambda committed:
python310Packages.fonttools: specify passthru.optional-dependencies

dotlambda committed:
python310Packages.pygal: update dependencies

SuperSandro2000 committed:
python310Packages.trio-websocket: init at 0.9.2

SuperSandro2000 committed:
python310Packages.selenium: 3.141.0 -> 4.2.0

mweinelt committed:
python3Packages.ansible-later: 2.0.13 -> 2.0.14

SuperSandro2000 committed:
python310Packages.splinter: disable by design not working test, remove six dependency

mweinelt committed:
python3Packages.audible: update httpx constraint, disable check phase

mweinelt committed:
python3Packages.sanic-tesing: relax httpx constraint

fabaff committed:
python310Packages.msgpack: 1.0.3 -> 1.0.4

mweinelt committed:
python3Packages.fastapi-mail: relax http constraint

mweinelt committed:
python3Packages.graph-tool: 2.43 -> 2.45

- Enable parallel build or else this monstrous build takes hours.
- Remove the maintainer who hasn't contributed to nixpkgs since 2016.
- Update dependencies

mweinelt committed:
python3Packages.sleekxmpp: disable on python3.10

The project has not been updated to work with Python 3.10, is archived
and recommends using slixmpp instead.

mweinelt committed:
python3Packages.homeconnect: propagate six

mweinelt committed:
python3Packages.aiosteamist: relax xmltodict constraint

mweinelt committed:
home-assistant: relax cryptography constraint

mweinelt committed:
python3Packages.cfn-flip: fix hash

mweinelt committed:
python3Packages.node_progressive: mark unconditionally as broken

mweinelt committed:
python3Packages.sphinxcontrib_bayesnet: mark unconditionally broken

mweinelt committed:
python3Packages.selectors2: fix mapping import

The library is otherwise unmaintained and should be removed once
something larger comes up.

mweinelt committed:
python3Packages.snowflake-connector-python: relax cryptography constraint

mweinelt committed:
quodlibet: 4.4.0 -> 4.5.0

https://quodlibet.readthedocs.io/en/latest/changelog.html#release-4-5-0

mweinelt committed:
diffoscope: 214 -> 215

Disable failing test on python3.10

SuperSandro2000 commented:
@fabaff https://hydra.nixos.org/build/179262332/nixlog/2

mweinelt commented:
I already fixed sanic-testing. Triggered a new eval just now.

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/205.html

mweinelt committed:
python3Packages.hyperion-py: fix python3.10 support

mweinelt committed:
python3Packages.requests: expose optional-dependencies

mweinelt committed:
python3Packages.beautifulsoup4: propagate chardet

Fixes the tests by making the package use chardet.

mweinelt committed:
gdown: propagate requests[socks]

mweinelt committed:
python3Packages.starlette: 0.19.0 -> 0.20.1

https://github.com/encode/starlette/releases/tag/0.20.1
https://github.com/encode/starlette/releases/tag/0.20.0
https://github.com/encode/starlette/releases/tag/0.19.1

mweinelt committed:
python3Packages.fastapi: 0.75.2 -> 0.78.0

https://github.com/tiangolo/fastapi/releases/tag/0.78.0
https://github.com/tiangolo/fastapi/releases/tag/0.77.1
https://github.com/tiangolo/fastapi/releases/tag/0.77.0
https://github.com/tiangolo/fastapi/releases/tag/0.76.0

mweinelt committed:
python3Packages.fastapi-mail: relax fastapi constraint

mweinelt commented:
> I already fixed sanic-testing. Triggered a new eval just now.

Oh yeah, it's broken *again*.

dotlambda committed:
mapnik: unvendor scons

dotlambda committed:
errbot: use python39

dotlambda committed:
ntfy: use python39 and remove ntfy-webpush from python3Packages

SuperSandro2000 committed:
python310Packages.limits: 2.6.2 -> 2.6.3

mweinelt committed:
cantoolz: fix python3.10 support

mweinelt committed:
i3a: relax python version constraint

mweinelt committed:
python3Packages.flickrapi: propagate six

mweinelt committed:
python3Packages.globus-sdk: fix tests

mweinelt committed:
python3Packages.snscrape: propagate requests[socks]

mweinelt committed:
turses: update tweepy override

mweinelt commented:
While borgbackup doesn't build right now, there is no need in fixing it. The 1.2.1 update allows msgpack==1.0.4, so it will be fixed once we merged into staging.

fabaff committed:
python310Packages.snscrape: unstable-2021-08-30 -> 0.4.3.20220106

fabaff committed:
python310Packages.websockets: 10.1 -> 10.3

fabaff committed:
python310Packages.unidecode: 1.3.2 -> 1.3.4

mweinelt committed:
python3Packages.snowflake-sqlalchemy: propgate six

fabaff committed:
python310Packages.databases: 0.5.5 -> 0.6.0

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/214.html

fabaff committed:
python310Packages.filelock: 3.6.0 -> 3.7.1

fabaff committed:
python310Packages.pybase64: 1.2.1 -> 1.2.2

fabaff committed:
python310Packages.immutables: 0.17 -> 0.18

SuperSandro2000 committed:
python310Packages.twine: 4.0.0 -> 4.0.1

fabaff committed:
python310Packages.pyathena: 2.5.2 -> 2.9.1

dotlambda committed:
privacyidea: use python39

dotlambda committed:
odoo: use python39

jonringer committed:
python3Packages.azure-mgmt-containerregistry: 9.1.0 -> 10.0.0

jonringer committed:
python3Packages.azure-storage-blob: 12.11.0 -> 12.12.0

jonringer committed:
python3Packages.azure-mgmt-containerinstance: 9.1.0 -> 9.2.0

fabaff committed:
python310Packages.glances-api: 0.3.5 -> 0.3.6

fabaff committed:
python310Packages.pymemcache: 3.5.1 -> 3.5.2

jonringer committed:
python3Packages.azure-data-tables: init at 12.4.0

jonringer committed:
python3Packages.antlr4-python3-runtime: fix tests for 4.9+

jonringer committed:
azure-cli: remove blanket python2 compat logic

jonringer committed:
azure-cli: 2.34.1 -> 2.37.0

SuperSandro2000 committed:
fixup! python3Packages.azure-data-tables: init at 12.4.0

SuperSandro2000 committed:
fixup! azure-cli: 2.34.1 -> 2.37.0

SuperSandro2000 committed:
python310Packages.pylint: 2.14.0 -> 2.14.1

fabaff committed:
python310Packages.pyserial: add pythonImportsCheck

- disable on obsolete Python releases
- add format

fabaff committed:
python310Packages.configobj: switch to pytestCheckHook

- add pythonImportsCheck
- disable on obsolete Python releases

fabaff committed:
python310Packages.jsonpatch: enable tests

- add pythonImportsCheck
- disable on obsolete Python releases

fabaff committed:
python310Packages.netifaces: add format

- disable on obsolete Python releases

fabaff committed:
python310Packages.pyannotate: switch to pytestCheckHook

- disable on obsolete Python releases
- add pythonImportsCheck

fabaff committed:
python310Packages.pytest-annotate: relax pytest constraint

fabaff committed:
pip-audit: adjust inputs

fabaff committed:
autoflake: switch to pytestCheckHook

- add pythonImportsCheck
- disable failing test

fabaff committed:
thrift: disable failing tests

mweinelt commented:
The `thrift` package, which is below `arrow-cpp` fails during the python test. cc @bjornfor 

```pytb
416/416 Test #416: python_test ......................***Failed    1.72 sec
Testing TBinaryAccelerated
Starting Benchmarks
HolyMoley Standard = 0.810447
HolyMoley Acceler. = 0.043564
FastStruct Standard = 0.185290
FastStruct Acceler. = 0.015597
Testing TCompactAccelerated
Traceback (most recent call last):
  File "/build/thrift-0.16.0/test/py/FastbinaryTest.py", line 255, in <module>
    do_test(TCompactProtocolAccelerated, TCompactProtocol)
  File "/build/thrift-0.16.0/test/py/FastbinaryTest.py", line 218, in do_test
    Test(fast, slow).do_test()
  File "/build/thrift-0.16.0/test/py/FastbinaryTest.py", line 184, in do_test
    self._check_read(rshuge)
  File "/build/thrift-0.16.0/test/py/FastbinaryTest.py", line 162, in _check_read
    c.read(prot)
  File "/build/thrift-0.16.0/build/test/py/gen-py-default/DebugProtoTest/ttypes.py", line 900, in read
    iprot._fast_decode(self, iprot, [self.__class__, self.thrift_spec])
SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats
*** FAILED ***
LIBDIR: /build/thrift-0.16.0/lib/py/build/lib.linux-x86_64-3.10
PY_GEN: gen-py-default
SCRIPT: FastbinaryTest.py
----------------
 Executing individual test scripts with various generated code directories
 Directories to be tested: gen-py-default, gen-py-slots, gen-py-oldstyle, gen-py-no_utf8strings, gen-py-dynamic, gen-py-dynamicslots
 Scripts to be tested: FastbinaryTest.py, TestFrozen.py, TestRenderedDoubleConstants.py, TSimpleJSONProtocolTest.py, SerializationTest.py, TestEof.py, TestSyntax.py, TestSocket.py
----------------

Testing script: /nix/store/ngxk330mm695j9gmrpzngj4v5fcdskzz-python3-3.10.4-env/bin/python3.10 /build/thrift-0.16.0/test/py/FastbinaryTest.py
----
Traceback (most recent call last):
  File "/build/thrift-0.16.0/test/py/RunClientServer.py", line 323, in <module>
    sys.exit(main())
  File "/build/thrift-0.16.0/test/py/RunClientServer.py", line 302, in main
    runScriptTest(options.libdir, options.gen_base, genpydir, script)
  File "/build/thrift-0.16.0/test/py/RunClientServer.py", line 103, in runScriptTest
    raise Exception("Script subprocess failed, retcode=%d, args: %s" % (ret, ' '.join(script_args)))
Exception: Script subprocess failed, retcode=1, args: /nix/store/ngxk330mm695j9gmrpzngj4v5fcdskzz-python3-3.10.4-env/bin/python3.10 /build/thrift-0.16.0/test/py/FastbinaryTest.py

fabaff committed:
python310Packages.qutip: 4.6.3 -> 4.7.0

fabaff committed:
python39Packages.rokuecp: disable failing tests

fabaff commented:
For now I disabled the `thrift` tests which fail on Python 3.10. The upstream repo contains hint that at least partially the tests were migrated to Python 3.10.

fabaff committed:
python310Packages.graphite-web: 1.1.8 -> 1.1.10

fabaff committed:
python310Packages.graphite-web: remove patch

fabaff committed:
calibre-web: add missing input

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/nix-macos-monthly/12330/24

Et7f3 commented on doc/languages-frameworks/python.section.md:
```diff
@@ -392,11 +395,11 @@ and adds it along with a `numpy` package to a Python environment.
 with import <nixpkgs> {};
 
 ( let
-    my_toolz = python38.pkgs.buildPythonPackage rec {
+    my_toolz = python39.pkgs.buildPythonPackage rec {
       pname = "toolz";
       version = "0.10.0";
 
-      src = python38.pkgs.fetchPypi {
+      src = python39.pkgs.fetchPypi {
         inherit pname version;
```
Their is a leftover in this expression `python38.withPackages (ps: [ps.numpy my_toolz])` Is it intentional ? Now It can be udpated to 3.10

mweinelt replied:
No, wasn't intentional. I apparently missed it.
